framework/src/st_things/things_stack/things_stack.c

```diff
@@ -385,11 +395,45 @@ int things_start_stack(void)
 		THINGS_LOG_E(TAG, "es_conn_type_Unknown");
 		return 0;
 	}
-
 	THINGS_LOG_V(TAG, "Stack Start Success");
 	return 1;
 }
 
+int things_start_scanning_ap() 
+{
+	pthread_mutex_lock(&g_things_scan_ap_mutex);
+	g_scan_ap_continue_flag = true;
+	pthread_mutex_unlock(&g_things_scan_ap_mutex);
+	if (pthread_create_rtos(&h_thread_things_scan_ap, NULL, auto_scanning_loop, NULL, THINGS_STACK_AP_SCAN_THREAD) != 0) {
+		THINGS_LOG_E(TAG, "Failed to create thread");
+		return 0;
+	}
+	return 1;
+}
+
+void things_stop_scanning_ap() 
+{
+	pthread_mutex_lock(&g_things_scan_ap_mutex);
+	g_scan_ap_continue_flag = false;
+	pthread_mutex_unlock(&g_things_scan_ap_mutex);
+}
+
+static void *__attribute__((optimize("O0"))) auto_scanning_loop(void)
+{
+	while (true) {
+		if (g_scan_ap_continue_flag == false)
+			break;
+		while (dm_is_es_complete() == false) {
```

Comment:
It is better to stop the scan after the device is connected with es_get_state().
es_complete returns true after being changed to STA and published to the cloud.
During the Easysetup operation, the ap scan operation is unnecessary.

Reply:
fixed